trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/remote/RemoteS3HostBuilder.java

```diff
@@ -13,7 +13,23 @@
  */
 package io.trino.aws.proxy.server.remote;
 
+import org.apache.velocity.VelocityContext;
+import org.apache.velocity.app.Velocity;
+
+import java.io.StringWriter;
+
 public interface RemoteS3HostBuilder
 {
+    static String getHostName(String bucket, String region, String domain, String hostnameTemplate)
+    {
+        VelocityContext context = new VelocityContext();
```

Comment:
Do we want to bring in Velocity just for this? It seems overkill to me. Every new dependency in a project is weight that must be maintained and a burden for users of the project. @vagaerg wdyt?

Reply:
IMO having templates makes it much easier to read and implement string evaluation. We could as well accomplish this using `String.replace` with additional code. I'm good either ways.

Reply:
I don't have a strong opinion. I originally suggested templating as I thought it'd be a cleaner implementation that would avoid some corner cases, but admittedly this is quite a simple usage of it.

Since our placeholders use `${}` notation (which are not valid characters in a domain) we should also be safe to use simpler logic rather than a fully fledged templating engine.

Thinking about this again, perhaps a simple string replacement is all we need for now - we can always revisit this later if we need more complex logic

Reply:
Let's use string formatting for this usecase since it can be accomplished without velocity